src/main/java/ch/cern/eam/wshub/core/services/contractmanagement/impl/EquipmentReservationAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+        addCustomerRentalAdjustment.setCustomerRentalAdjustment(tools.getInforFieldTools()
+                .transformWSHubObject(new net.datastream.schemas.mp_entities.customerrentaladjustment_001.CustomerRentalAdjustment(), validateCreateObject(equipmentReservationAdjustment), context));
+        return addCustomerRentalAdjustment;
+    }
+
+    private MP7863_GetCustomerRentalAdjustment_001 createGetOperation(String number) {
+        MP7863_GetCustomerRentalAdjustment_001 getCustomerRentalAdjustment = new MP7863_GetCustomerRentalAdjustment_001();
+        CUSTOMERRENTALADJUSTMENTID_Type idType = new CUSTOMERRENTALADJUSTMENTID_Type();
+        idType.setCUSTOMERRENTALADJUSTMENTPK(number);
+        getCustomerRentalAdjustment.setCUSTOMERRENTALADJUSTMENTID(idType);
+        return getCustomerRentalAdjustment;
+    }
+
+    private MP7865_SyncCustomerRentalAdjustment_001 createUpdateOperation(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7865_SyncCustomerRentalAdjustment_001 syncCustomerRentalAdjustment = new MP7865_SyncCustomerRentalAdjustment_001();
+        syncCustomerRentalAdjustment.setCustomerRentalAdjustment(tools.getInforFieldTools()
+                .transformWSHubObject(new net.datastream.schemas.mp_entities.customerrentaladjustment_001.CustomerRentalAdjustment(), validateUpdateObject(equipmentReservationAdjustment), context));
+        return syncCustomerRentalAdjustment;
+    }
+
+    private MP7866_DeleteCustomerRentalAdjustment_001 createDeleteOperation(String number) {
+        MP7866_DeleteCustomerRentalAdjustment_001 deleteCustomerRentalAdjustment = new MP7866_DeleteCustomerRentalAdjustment_001();
+        CUSTOMERRENTALADJUSTMENTID_Type idType = new CUSTOMERRENTALADJUSTMENTID_Type();
+        idType.setCUSTOMERRENTALADJUSTMENTPK(number);
+        deleteCustomerRentalAdjustment.setCUSTOMERRENTALADJUSTMENTID(idType);
+        return deleteCustomerRentalAdjustment;
+    }
+
+    private EquipmentReservationAdjustment validateCreateObject(EquipmentReservationAdjustment equipmentReservationAdjustment) {
+        validateUpdateObject(equipmentReservationAdjustment);
+        if (isEmpty(equipmentReservationAdjustment.getCode())) {
+            equipmentReservationAdjustment.setCode("1");
+        }
+        return equipmentReservationAdjustment;
+    }
+
+    private EquipmentReservationAdjustment validateUpdateObject(EquipmentReservationAdjustment equipmentReservationAdjustment) {
+        if (isEmpty(equipmentReservationAdjustment.getStatusRCode())) {
+            equipmentReservationAdjustment.setStatusRCode(equipmentReservationAdjustment.getStatusCode());
+        }
+        if (isEmpty(equipmentReservationAdjustment.getTypeRCode())) {
+            equipmentReservationAdjustment.setTypeRCode(equipmentReservationAdjustment.getTypeCode());
+        }
+        return equipmentReservationAdjustment;
+    }
```

Comment:
This is probably not needed on update, so no need to have this dedicated method.